db-service/lib/cqn2sql.js

```diff
@@ -435,15 +435,16 @@ class CQN2SQLRenderer {
       throw cds.error`Cannot insert rows without columns or elements`
     }
     let columns = INSERT.columns || (elements && ObjectKeys(elements))
+    const columnIndex = columns.reduce((m, c, i) => ((m[c] = i), m), {})
     if (elements) {
       columns = columns.filter(c => c in elements && !elements[c].virtual && !elements[c].isAssociation)
     }
```

Comment:
Isn't the problem stemming from here? 
If we are called with a specified list of columns then this list of columns has to be valid or lead to an error, but not silent skipping → we should simply remove these lines imho instead 
```suggestion
```